src/MultiObjectiveAlgorithms.jl

```diff
@@ -19,11 +19,14 @@ function Base.isapprox(a::SolutionPoint, b::SolutionPoint; kwargs...)
     return isapprox(a.y, b.y; kwargs...)
 end
 
+Base.:(==)(a::SolutionPoint, b::SolutionPoint) = a.y == b.y
+
 function filter_nondominated(solutions::Vector{SolutionPoint})
-    sort!(solutions; by = x -> x.y)
+    solutions = sort(solutions; by = x -> x.y)
     nondominated_solutions = SolutionPoint[solutions[1]]
     for i in 2:length(solutions)
-        if !(nondominated_solutions[end] ≈ solutions[i])
+        objective_vectors = zip(nondominated_solutions[end].y, solutions[i].y)
+        if !any(x -> x[1] ≈ x[2], objective_vectors)
```

Comment:
I am sorry but I do not get this. To filter weakly non-dominated solutions we need to take the ObjectiveSense into account.

Reply:
In case of MIN
```julia
if any(x -> x[1] <= x[2], objective_vectors)
```
seems right for 2-objectives.

However for 3 or more, it is not sufficient to compare the i-th solution with the last nondominated one.

Example in MIN:
```julia
a = [0,1,1]
b = [0,2,0]
c = [1,1,1]
```
were `a <= c`

Reply:
Fixed and added this as a test.